qiskit_nature/circuit/library/initial_states/slater_determinants.py

```diff
@@ -0,0 +1,104 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Slater determinants."""
+
+from typing import Optional
+
+import numpy as np
+from qiskit import QuantumCircuit, QuantumRegister
+from qiskit_nature.converters.second_quantization import QubitConverter
+from qiskit_nature.mappers.second_quantization import JordanWignerMapper
+
+from .utils.givens_rotations import _prepare_slater_determinant_jordan_wigner
+
+
+def _rows_are_orthonormal(mat: np.ndarray, rtol: float = 1e-5, atol: float = 1e-8) -> bool:
+    m, _ = mat.shape
+    return np.allclose(mat @ mat.T.conj(), np.eye(m), rtol=rtol, atol=atol)
+
+
+class SlaterDeterminant(QuantumCircuit):
+    r"""A circuit that prepares a Slater determinant.
+
+    A Slater determinant is a state of the form
+
+    .. math::
+        b^\dagger_1 \cdots b^\dagger_{N_f} \lvert \text{vac} \rangle
+
+    where
+
+    .. math::
+        b^\dagger_j = \sum_{k = 1}^N Q_{jk} a^\dagger_k,
+
+    - :math:`Q` is an :math:`N_f \times N` matrix with orthonormal rows
```

Comment:
Compared to the explanation of `W` in the `FermionicGaussianState`, the explanation of `Q` is lacking. Maybe you can again add a section which explains how this matrix can be obtained further down in this docstring.

Reply:
I didn't mention QuadraticHamiltonian here because I think the most common way to obtain `Q` will be from electronic structure codes. I will add that to the tutorial.

Reply:
On further thought, I don't think `Q` will be obtained from electronic structure codes because they output the optimized molecular orbitals directly and the initial guess is not important. I've updated the docstring to mention how to obtain the matrix using the QuadraticHamiltonian class.